pkg/cmd/printer/printer.go

```diff
@@ -487,3 +496,72 @@ func (p *ignoreEventPrinter) Print(event trace.Event) {}
 func (p *ignoreEventPrinter) Epilogue(stats metrics.Stats) {}
 
 func (p ignoreEventPrinter) Close() {}
+
+// forwardEventPrinter sends events over the Fluent Forward protocol to a receiver
+type forwardEventPrinter struct {
+	destination string
+	tag         string
+	client      *forward.Client
+}
+
+func (p *forwardEventPrinter) Init() error {
+	// Set up a default tag if required
+	if p.tag == "" {
+		p.tag = "tracee"
+	}
+
+	// Create a TCP connection to the forward receiver
+	p.client = forward.New(forward.ConnectionOptions{
+		Factory: &forward.ConnFactory{
+			Address: p.destination,
+		},
+	})
+	if err := p.client.Connect(); err != nil {
+		return fmt.Errorf("error connecting to Forward destination %q: %v", p.destination, err)
+	}
+	return nil
+}
+
+func (p *forwardEventPrinter) Preamble() {}
+
+func (p *forwardEventPrinter) Print(event trace.Event) {
+	if p.client == nil {
+		logger.Error("invalid Forward client")
+	}
+
+	// The actual event is marshalled as JSON then sent with the other information (tag, etc.)
+	eBytes, err := json.Marshal(event)
+	if err != nil {
+		logger.Error("error marshaling event to json", "error", err)
+	}
+
+	record := map[string]interface{}{
+		"event": string(eBytes),
+	}
+
+	err = p.client.SendMessage(p.tag, record)
+	// Assuming all is well we continue but if the connection is dropped or some other error we retry
+	if err != nil {
+		logger.Error("error writing to Forward destination", "url", p.destination, "tag", p.tag, "error", err)
+		// Try five times to reconnect and send before giving up
+		for attempts := 0; attempts < 5; attempts++ {
```

Comment:
'5' is a magic number, potentially we expose via an option.

We will get reports of error writing until the destination appears again.

Reply:
not for this PR, but can you add a TODO so we consider using go-kit for circuit break, retry, etc

Reply:
Do we already do that anywhere btw? I can adopt if so.

Reply:
@patrick-stephens we don't, if you want to do it, that would great but perhaps on a next PR, let's wrap this one up first